packages/quantic/force-app/main/default/lwc/quanticCarousel/quanticCarousel.js

```diff
@@ -0,0 +1,145 @@
+import nextPage from '@salesforce/label/c.quantic_NextPage';
+import previousPage from '@salesforce/label/c.quantic_PreviousPage';
+import propertyIsRequired from '@salesforce/label/c.quantic_PropertyIsRequired';
+import {I18nUtils} from 'c/quanticUtils';
+import {LightningElement, api} from 'lwc';
+
+/**
+ * The `QuanticCarousel` component displays the content provided in a carousel. This component also displays navigation buttons that allow the end user to navigate between the different pages of the carousel.
+ * @category Utility
+ * @example
+ * <c-quantic-carousel number-of-pages="2" number-of-items-per-page="1">
+ *   <div slot="carousel-item">Item One</div>
+ *   <div slot="carousel-item">Item Two</div>
+ * </c-quantic-carousel>
+ */
+export default class QuanticCarousel extends LightningElement {
+  labels = {
+    propertyIsRequired,
+    invalidPositiveIntegerProperty,
+    nextPage,
+    previousPage,
+    goToPage,
+  };
+
+  /**
+   * The total number of pages.
+   * @api
+   * @type {number}
+   */
+  @api numberOfPages;
+  /**
+   * The number of items to display per page.
+   * @api
+   * @type {number}
+   */
+  @api numberOfItemsPerPage;
+  /**
+   * The index of the current page.
+   * @api
+   * @type {number}
+   */
+  @api
+  get currentPage() {
+    return this._currentPage;
+  }
+  set currentPage(value) {
+    this._currentPage = value;
+  }
+
+  /** @type {number} */
+  _currentPage = 0;
+  /** @type {string} */
+  errorMessage = '';
+
+  connectedCallback() {
+    this.validateProperties();
+  }
+
+  renderedCallback() {
+    this.hideNonVisibleSlides();
+  }
+
+  validateProperties() {
+    if (!this.numberOfPages || !this.numberOfItemsPerPage) {
+      this.setInitializationError(
+        I18nUtils.format(
+          this.labels.propertyIsRequired,
+          !this.numberOfPages ? 'numberOfPages' : 'numberOfItemsPerPage'
+        )
+      );
+    } else if (isNaN(this.numberOfPages) || isNaN(this.numberOfItemsPerPage)) {
+      this.setInitializationError(
+        I18nUtils.format(
+          this.labels.invalidPositiveIntegerProperty,
+          isNaN(this.numberOfPages) ? 'numberOfPages' : 'numberOfItemsPerPage'
+        )
+      );
+    }
+  }
+
+  hideNonVisibleSlides() {
+    /** @type {HTMLSlotElement} */
+    const slot = this.template.querySelector('slot[name="carousel-item"]');
+    const items = slot?.assignedNodes();
+
+    items?.forEach((item, index) => {
+      const isDisplayedInCurrentPage =
+        Math.floor(index / this.numberOfItemsPerPage) === this._currentPage;
+      if (!isDisplayedInCurrentPage) {
+        // @ts-ignore
+        item.style.display = 'none';
+      } else {
+        // @ts-ignore
+        item.style.display = '';
+      }
+    });
+  }
+
+  handlePrevious() {
+    // eslint-disable-next-line @lwc/lwc/no-api-reassignments
+    this.currentPage = this._currentPage - 1;
+  }
+
+  handleNext() {
```

Comment:
Maybe I missed it but isnt the previous/next buttons and indicators supposed to dispatch a custom `pageChange` event?

Reply:
that was the initial plan, but as discussed in the last meeting we has about the carousel component, we will send any events. all the logic to move next/previous or to any carousel page will be encapsulated and isolated in the Quantic Carousel component.